lib/TPP/MapToBatchReduceGEMM.cpp

```diff
@@ -126,6 +200,53 @@ getSlicedOperands(OpBuilder &builder, Location loc, ValueRange localIvs,
   return slicedOperands;
 }
 
+// Returns true if linalg op is using its second operand in VNNI format
+static bool isLinalgOpVNNI(linalg::LinalgOp linalgOp) {
+  if (linalgOp.getNumLoops() == 7 && linalgOp->getOperands().size() >= 3 &&
+      linalgOp->getOperands()[1]
+          .getType()
+          .cast<ShapedType>()
+          .getElementType()
+          .isBF16() &&
+      linalgOp->getOperands()[1]
+              .getType()
+              .cast<ShapedType>()
+              .getShape()[linalgOp->getOperands()[1]
+                              .getType()
+                              .cast<ShapedType>()
+                              .getRank() -
+                          1] ==
+          vnni::utils::getVNNIBlockingFactor(
+              linalgOp->getOperands()[1].getType())) {
+    if (!linalgOp.hasBufferSemantics()) {
```

Comment:
this check is too restrictive. Looking for a pack may not be right, if you run this after propagation. The pack cannot be there anymore.

Reply:
I'll get rid of this.